src/Microsoft.AspNetCore.Authentication/ClaimsTransformationMiddleware.cs

```diff
@@ -39,13 +39,18 @@ public class ClaimsTransformationMiddleware
             handler.RegisterAuthenticationHandler(context.GetAuthentication());
             try
             {
-                if (Options.Transformer != null)
+                var transform = Options.Transformer;
+                if (transform == null && Options.TransformerType != null)
+                {
+                    transform = context.RequestServices.GetService(Options.TransformerType) as IClaimsTransformer;
```

Comment:
Any particular reason not to use `GetRequiredService`? Using `GetService` + an as cast will "fail" silently and won't help one understand what's wrong with the DI registration.